src/main/java/graphql/execution/AsyncExecutionStrategy.java

```diff
@@ -45,25 +45,23 @@ public CompletableFuture<ExecutionResult> execute(ExecutionContext executionCont
         ExecutionStrategyInstrumentationContext executionStrategyCtx = ExecutionStrategyInstrumentationContext.nonNullCtx(instrumentation.beginExecutionStrategy(instrumentationParameters, executionContext.getInstrumentationState()));
 
         MergedSelectionSet fields = parameters.getFields();
-        Set<String> fieldNames = fields.keySet();
+        List<String> fieldNames = fields.getKeys();
         Async.CombinedBuilder<FieldValueInfo> futures = Async.ofExpectedSize(fields.size());
```

Comment:
the implementation is ImmutabeList.copyOf and hence backed by the original structure